ipaserver/install/plugins/update_ra_cert_store.py

```diff
@@ -2,74 +2,92 @@
-        if not olddb.has_nickname('ipaCert'):
+        certdb = NSSDatabase(nssdir=paths.HTTPD_ALIAS_DIR)
+        if not certdb.has_nickname(ra_nick):
             # Nothign to do
             return False, []
 
-        newdb = certdb.NSSDatabase(nssdir=paths.IPA_RADB_DIR)
-        if os.path.exists(paths.IPA_RADB_DIR):
-            if newdb.has_nickname('ipaCert'):
+        if os.path.exists(paths.RA_AGENT_PEM):
+            if certdb.has_nickname(ra_nick):
                 self.log.warning(
-                    "An 'ipaCert' nickname exists in both the old {} and the "
-                    "new {} NSS Databases!".format(paths.HTTPD_ALIAS_DIR,
-                                                   paths.IPA_RADB_DIR))
+                    "A certificate with the nickname 'ipaCert' exists in "
+                    "the old '{}' NSS database as well as in the new "
+                    "PEM file '{}'"
+                    .format(paths.HTTPD_ALIAS_DIR, paths.RA_AGENT_PEM))
                 return False, []
-        else:
-            # Create the DB
-            newdb.create_db(user=IPAAPI_USER, group=IPAAPI_GROUP, backup=True)
-
-        # Import cert chain (ignore errors, as certs may already be imported)
-        certlist = olddb.list_certs()
-        certflags = {}
-        for name, flags in certlist:
-            certflags[name] = flags
-        for name in olddb.get_trust_chain('ipaCert'):
-            if name == 'ipaCert':
-                continue
-            try:
-                cert = olddb.get_cert(name, pem=True)
-                newdb.add_cert(cert, name, certflags[name], pem=True)
-            except Exception as e:  # pylint disable=broad-except
-                self.log.warning("Failed to import '{}' from trust "
-                                 "chain: {}".format(name, str(e)))
-
-        # As the last step export/import/delete the RA Cert
-        pw = binascii.hexlify(os.urandom(10))
-        p12file = os.path.join(paths.IPA_RADB_DIR, 'ipaCert.p12')
-        olddb.export_pkcs12('ipaCert', p12file, pw)
-        newdb.import_pkcs12(p12file, pw)
-
-        certmonger.stop_tracking(secdir=olddb.secdir,
-                                 nickname='ipaCert')
-        certmonger.start_tracking(secdir=newdb.secdir,
-                                  nickname='ipaCert',
-                                  password_file=newdb.pwd_file)
-
-        olddb.delete_cert('ipaCert')
+
+        p12file = tempfile.mkstemp(dir=certdb.secdir)
+        # no password is necessary as we will be saving it in clear anyway
+        certdb.export_pkcs12(ra_nick, p12file)
+
+        # get the private key and certificate from the file
+        with open(p12file, 'r') as f:
+            data = f.read()
+            cert = crypto_x509.load_pem_x509_certificate(
+                data, default_backend())
+            key = serialization.load_pem_private_key(
+                data, password=None, backend=default_backend())
+        os.remove(p12file)
```

Comment:
I was actually able to simplify the code very much since the same as here is what happens during DL0 replica installation therefore `CAInstance.import_ra_cert()` was already prepared for this.